### PR TITLE
1.0 API

### PR DESCRIPTION
**Pull request by ffuuugor**
We're introducing new Opacus API. See Readme and updated turorials for details.

For code reviews and discussions see PR history https://github.com/pytorch/opacus/pulls?q=is%3Apr+is%3Aclosed+base%3Aexperimental_v1.0

### CODE REVIEW

**Comment by karthikprasad** [comment]
Support for 3.6.x was added to be colab friendly. Colab now supports 3.7. I propose we just remove support for 3.6 to make the tests happy.

File: .github/workflows/test-installation.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       max-parallel: 48
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
-        python-version: [3.7.4, 3.7.12, 3.8, 3.9]
+        python-version: [3.7.5, 3.7.12, 3.8, 3.9]
```
Comment (karthikprasad) [inline .github/workflows/test-installation.yml:18]: oh. Why? Doesn't windows or mac support 3.7.4?

Comment (ffuuugor) [reply .github/workflows/test-installation.yml:18]: https://github.com/pytorch/opacus/runs/4327225058?check_suite_focus=true

Comment (ffuuugor) [reply .github/workflows/test-installation.yml:18]: Ubuntu doesn't, and I think it tries all combinations of os/python/pytorch versions

**Review by karthikprasad — APPROVED** [review]

**Comment by karthikprasad** [comment]
FTR: #293, #294, #295, #296, and #298 are also part of this release despite landing after this PR.

**Pull request merged**